packages/wallets/src/features/cashier/modules/WithdrawalCrypto/provider/WithdrawalCryptoProvider.tsx

```diff
@@ -76,6 +90,16 @@ const WithdrawalCryptoProvider: React.FC<React.PropsWithChildren<TWithdrawalCryp
         return () => unsubscribe();
     }, [activeWallet?.currency, activeWallet?.loginid, subscribe, unsubscribe]);
 
+    useEffect(() => {
+        if (verificationCode) {
```

Comment:
is there a need to call the API call using a useEffect within a provider?

Reply:
feels a bit like a specific component concern to me, I can imagine the dependencies changing for random reason (e.g. mutateAsync changing because reconnection happen). 

generally, we should avoid driving actions based on side effect derived from other, not-always-related variables changing, better to use specific events derived e.g. form compnent lifecycle

Reply:
hey @shahzaib-deriv @wojciech-deriv 
this useEffect should be fired on the first mount of WithdrawalCrypto module to check that the user doesn't use the same verification code to withdraw the funds.
feel free to suggest other approach to handle this 🙏